### ISSUE TITLE
Indicate and warn when an inherited scene is using the base scene's script

### ISSUE THREAD

**Issue by Error7Studios** [godotengine/godot-proposals#3016]
### Describe the project you are working on

Applies to any project using inherited scenes.

### Describe the problem or limitation you are having in your project

When creating an inherited scene, there's no visual indication as to whether or not the base scene's script has been replaced.
It is often desired to give the inherited scene a new script which extends the base scene's script.

It is very easy to forget to change this script.
Worse, there's a risk of clicking on that script and making changes to it, thinking you are doing this on a new script, but actually are making changes to the base script.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Indicate that a script is inherited from a base scene by greying out the script icon (like with inherited nodes).
Optionally, also add a node configuration warning like:
```
The attached script is inherited from the base scene.
Consider adding a new script which extends the base scene's script.
```

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![Inherited Script Warning](https://user-images.githubusercontent.com/60025044/126275307-8185781a-6b80-468c-9626-a83fe5d2333c.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

This changes built in `SceneTree` behavior.

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot-proposals/issues/2976.

**Comment by LikeLakers2** [comment]
I don't see any reason to add a configuration warning when there's not really anything that could be considered wrong. Sometimes, you *want* an inherited scene to have an inherited script, and that's okay.

However, I still see the concern -- and so I think it'd be reasonable to have the script icon grayed out, if the script is a inherited one.

Still, we *could* warn the user that a script is a inherited script if they click on the grayed script icon. We could then offer a couple options within the warning: Edit the original script, or Create a new script inheriting from the original script.

**Comment by CedNaru** [comment]
Yeah, nothing wrong or right about changing the original scripts. Every inherited scene exists for differents changes that can have nothing to do with scripts.
I think the more generalized idea is just to make more obvious what has been changed in the inherited scene compared to the base one.

**Comment by Error7Studios** [comment]
> I think the more generalized idea is just to make more obvious what has been changed in the inherited scene compared to the base one.

How would be best to do that? Putting an icon next to the node that has changes from the base scene?
If so, maybe the user can click on that icon, and a window will pop up showing the changed property names.

The inspector for that node could also have a new section at the top, that shows those changed properties.
(They shouldn't be removed from their original section, though, just show a duplicate property at the top)
